src/SFA.DAS.Commitments.Notification.WebJob/EmailServices/SendingEmployerTransferRequestEmailService.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Polly;
+using Polly.Retry;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities;
+using SFA.DAS.Commitments.Notification.WebJob.Models;
+using SFA.DAS.EAS.Account.Api.Client;
+using SFA.DAS.EAS.Account.Api.Types;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Notifications.Api.Types;
+
+namespace SFA.DAS.Commitments.Notification.WebJob.EmailServices
+{
+    public class SendingEmployerTransferRequestEmailService : ISendingEmployerTransferRequestEmailService
+    {
+        private readonly ICommitmentRepository _commitmentRepository;
+        private readonly IAccountApiClient _accountApi;
+        private readonly ILog _logger;
+        private readonly RetryPolicy _retryPolicy;
+
+        public SendingEmployerTransferRequestEmailService(ICommitmentRepository commitmentRepository,
+            IAccountApiClient accountApi,
+            ILog logger)
+        {
+            _commitmentRepository = commitmentRepository;
+            _accountApi = accountApi;
+            _logger = logger;
+            _retryPolicy = GetRetryPolicy();
+        }
+
+        public async Task<IEnumerable<Email>> GetEmails()
+        {
+            var transferRequests = await _commitmentRepository.GetPendingTransferRequests();
+
+            _logger.Info($"Found {transferRequests.Count} pending transfer requests");
+
+            var distinctEmployerAccountIds = transferRequests.Select(x => x.SendingEmployerAccountId)
+                .Distinct()
+                .ToList();
+
+            var distinctAccountsTasks =
+                distinctEmployerAccountIds
+                    .Select(x => _retryPolicy.ExecuteAndCaptureAsync(() => _accountApi.GetAccount(x)))
+                    .ToList();
+
+            var userPerAccountTasks =
+                distinctEmployerAccountIds
+                    .Select(ToUserModel)
+                    .ToList();
+
+            await Task.WhenAll(distinctAccountsTasks);
+            await Task.WhenAll(userPerAccountTasks);
+
+            var accounts = distinctAccountsTasks
```

Comment:
If any of the tasks fail, then the WhenAll call will throw an Exception, so there's no need to check for successful results. Instead, we could combine the two lines into this...
 var accounts = await Task.WhenAll(distinctAccountsTasks);